src/editor/action-executor.ts

```diff
@@ -412,6 +437,42 @@ export class ActionExecutor {
                 break;
             }
 
+            case EditActionType.InsertDotMethod: {
+                const initialBoundary = this.getBoundaries(context.expressionToLeft);
+                const root = context.expressionToLeft.rootNode as Statement;
+                const index = context.expressionToLeft.indexInRoot;
+
+                const newCode = new ExprDotMethodStmt(
+                    action.data.functionName,
+                    action.data.args,
+                    action.data.returns,
+                    action.data.exprType
+                );
+
+                const replacementType = context.expressionToLeft.canReplaceWithConstruct(newCode);
+
+                if (replacementType !== InsertionType.Invalid) {
+                    this.module.closeConstructDraftRecord(root.tokens[index]);
+
+                    newCode.setExpression(context.expressionToLeft);
+                    newCode.indexInRoot = index;
+                    newCode.rootNode = root;
+                    root.tokens[index] = newCode;
+                    root.rebuild(root.getLeftPosition(), 0);
+
+                    this.module.editor.executeEdits(initialBoundary, newCode);
+                    this.module.focus.updateContext(newCode.getInitialFocus());
+
+                    if (replacementType !== InsertionType.DraftMode) {
+                        this.module.closeConstructDraftRecord(context.expressionToLeft);
```

Comment:
I think line 455 does this already, or is there a reason to do it again here? Aren't `root.tokens[index]` is the same as `context.expressionToLeft` since `index = context.expressionToLeft.indexInRoot`?

Reply:
yeah, you're right. fixed this.